integration-cli/docker_cli_build_test.go

```diff
@@ -6740,3 +6740,37 @@ func (s *DockerSuite) TestBuildLabelsOverride(c *check.C) {
 	}
 
 }
+
+// #20083
+func (s *DockerSuite) TestBuildDockerignoreComment(c *check.C) {
+	name := "testbuilddockerignorecleanpaths"
+	dockerfile := `
+        FROM busybox
+        ADD . /tmp/
+        RUN sh -c "(! ls /tmp/foo) && (! ls /tmp/foo2) && (ls /tmp/dir1/foo) && (ls /tmp/#1) && (! ls /tmp/#2)"`
+	ctx, err := fakeContext(dockerfile, map[string]string{
+		"foo":      "foo",
+		"foo2":     "foo2",
+		"dir1/foo": "foo in dir1",
+		"#1":       "# file 1",
+		"#2":       "# file 2",
+		".dockerignore": `# Visual C++ cache files
+# because we have git ;-)
+# The above comment is from #20083
+./foo
+#dir1/foo
+./foo2
+# The following is considered as comment as # is at the beginning
+#1
+# The following is not considered as comment as # is not at the beginning
+  #2
```

Comment:
LOL I guess, but I think that's less likely than leading spaces for comments, which I know people dislike about Dockerfiles.  I just don't see the benefit of keeping this odd behavior. Eventually Dockerfile processing will be changed, but these .dockerignore files will probably live on longer after Dockerfiles are "done right".  I'd suggest we do it right for these.